service/inspectImages.go

```diff
@@ -0,0 +1,271 @@
+		}
+		t, image := validateAndExtractImage(im)
+		if !t {
+			return errors.New("image not valid")
+		}
+		appendImage(image, &i.buffer)
+	}
+	os.RemoveAll(folder)
+	return nil
+}
+
+func validateAndExtractImage(s string) (bool, image) {
+	var rex = regexp.MustCompile(`^(((?P<o>[A-Za-z0-9_\-.:]+)\/){1,2})?((?P<i>[A-Za-z0-9_\-.]+)\:){1}(?P<v>[A-Za-z0-9_\-.]+){1}`)
+	var im image
+	if rex.MatchString(s) {
+		regexResult := rex.FindAllStringSubmatch(s, -1)[0]
+		h := strings.Split(regexResult[1], "/")
+		hostname := ""
+		if len(h) > 2 {
+			hostname = h[0]
+		}
+		im = image{Hostname: hostname, Organization: regexResult[3], Image: regexResult[5], Version: regexResult[6]}
+		return true, im
+	}
+	return false, im
+}
+
+func (i *ImagesService) extractFiles(fileName string) (string, error) {
+	folder, err := ioutil.TempDir("", "tar")
+	if err != nil {
+		i.logger.Error("creating temp dir %s: %s", folder, err)
+		return "", err
+	}
+
+	cmd := exec.Command("tar", "-xf", fileName, "-C", folder, "--strip-components=1")
+	err = cmd.Run()
+	if err != nil {
+		i.logger.Error("running extract command for %s: %s", fileName, err)
+		os.RemoveAll(folder)
+		return "", err
+	}
+	return folder, nil
+}
+
+func applyTemplateToImage(s string, m map[string]interface{}, l utils.Logger) (string, error) {
+	s, err := renderTemplate(s, m, l)
+	if err != nil {
+		l.Error("rendering template %s: %s", s, err)
+		return "", err
+	}
+	return s, nil
+}
+
+func renderTemplate(temp string, m map[string]interface{}, l utils.Logger) (string, error) {
+	tmpl, err := template.New("new").Parse(strings.Replace(temp, ".Values", "", 4))
+	if err != nil {
+		l.Error("processing template %s: %s", temp, err)
+		return temp, err
+	}
+	b := bytes.NewBufferString("")
+	err = tmpl.Execute(b, m)
+	if err != nil {
+		l.Error("rendering template %s: %s", temp, err)
+		return temp, err
+	}
+	return b.String(), nil
+}
+
+func appendImage(i image, buffer *bytes.Buffer) {
+	if i.Hostname == "" && i.Organization == "" && i.Image != "" && i.Version != "" {
+		s := fmt.Sprintf("%s:%s\n", i.Image, i.Version)
+		buffer.WriteString(s)
+	} else if i.Hostname == "" && i.Organization != "" && i.Image != "" && i.Version != "" {
+		s := fmt.Sprintf("%s/%s:%s\n", i.Organization, i.Image, i.Version)
+		buffer.WriteString(s)
+	} else if i.Hostname != "" && i.Organization != "" && i.Image != "" && i.Version != "" {
+		s := fmt.Sprintf("%s/%s/%s:%s\n", i.Hostname, i.Organization, i.Image, i.Version)
+		buffer.WriteString(s)
+	}
+}
+
+func extractImagesTemplates(shell string, l utils.Logger) ([]string, error) {
+	cmd := exec.Command(shell, "-c", "ls *.yaml | xargs -i grep 'image:' {}")
+	out, err := cmd.Output()
+	if err != nil && !strings.Contains(err.Error(), "123") {
+		l.Error("reading yaml files: %s", err)
+		return nil, err
+	}
+	var images []string
+	for _, i := range strings.Split(string(out), "\n") {
+		if i != "" {
+			images = append(images, cleanupImageString(i))
+		}
+	}
+	return images, nil
```

Comment:
Please, let's not shell out for that.

The rendered template is a YAML file, let's handle that as a YAML, aka just a dictionary. Let's iterate over it looking for keys with value `image`.